GHA: Dispatch event requires PAT

### Description
<!-- A concise description of why this chore matters, who will enjoy it and how. -->

Release dispatch fails in GHA due to [known restrictions](https://github.com/peter-evans/repository-dispatch#token).
```text
delivery / release-dispatch / release-dispatch
failed 6 minutes ago in 6s
Search logs
  Set up job3s
  Repository Dispatch to Docs3s
    client-payload: {}
Run peter-evans/repository-dispatch@v1
  with:
    token: ***
    event-type: pack-release
    repository: buildpacks/docs
    client-payload: {}
##[error]Resource not accessible by integration
  Complete job
```

### Proposed solution
<!-- A clear and concise description of how do you think the chore should be implemented. -->
As proposed in the GHA we are using for dispatching, we should be using a [PAT](https://help.github.com/en/github/authenticating-to-github/creating-a-personal-access-token-for-the-command-line) with appropriate permissions.


### Additional context
<!-- Add any other context or screenshots about the chore that may help. -->

We have a "bot" user @buildpack-bot that we can use for this integration.

--- a/.github/workflows/delivery-homebrew.yml
+++ b/.github/workflows/delivery-homebrew.yml
@@ -15,8 +15,9 @@ jobs:
         with:
           repository: buildpack/homebrew-tap
           path: homebrew-tap
+          token: ${{ secrets.BOT_TOKEN }}
       - name: Copy pack.rb
-        run: cp .github/workflows/delivery/homebrew/pack.rb homebrew-tap/Formula/pack.rb 
+        run: cp .github/workflows/delivery/homebrew/pack.rb homebrew-tap/Formula/pack.rb
       - name: Lookup assets
         uses: actions/github-script@v1
         id: assets
@@ -39,7 +40,7 @@ jobs:
           curl -sSL ${{ steps.assets.outputs.linux_url }} -o ${{ steps.assets.outputs.linux_name }}
           linux_sha256=$(sha256sum ${{ steps.assets.outputs.linux_name }} | cut -d ' ' -f1)
           echo "::set-output name=linux_sha256::$linux_sha256"
-          
+
           curl -sSL ${{ steps.assets.outputs.macos_url }} -o ${{ steps.assets.outputs.macos_name }}
           macos_sha256=$(sha256sum ${{ steps.assets.outputs.macos_name }} | cut -d ' ' -f1)
           echo "::set-output name=macos_sha256::$macos_sha256"
@@ -57,9 +58,9 @@ jobs:
       - run: cat homebrew-tap/Formula/pack.rb
       - name: Commit changes
         run: |
-          git config --global user.name "github-bot"
-          git config --global user.email "action@github.com"
-          
+          git config --global user.name "buildpack-bot"
+          git config --global user.email "cncf-buildpacks-maintainers@lists.cncf.io"
+
           cd homebrew-tap
           git add Formula/pack.rb
           git commit -m "Version ${{ github.event.release.tag_name }}"

--- a/.github/workflows/delivery-release-dispatch.yml
+++ b/.github/workflows/delivery-release-dispatch.yml
@@ -6,12 +6,12 @@ on:
       - published
 
 jobs:
-  release-dispatch:
+  dispatch2docs:
     runs-on: ubuntu-latest
     steps:
       - name: Repository Dispatch to Docs
         uses: peter-evans/repository-dispatch@v1
         with:
-          token: ${{ secrets.GITHUB_TOKEN }}
+          token: ${{ secrets.BOT_TOKEN }}
           event-type: pack-release
           repository: buildpacks/docs

--- a/.github/workflows/testdata/event-release.json
+++ b/.github/workflows/testdata/event-release.json
@@ -1,15 +1,15 @@
 {
   "action": "published",
   "release": {
-    "url": "https://api.github.com/repos/buildpacks/pack/releases/26958076",
-    "assets_url": "https://api.github.com/repos/buildpacks/pack/releases/26958076/assets",
-    "upload_url": "https://uploads.github.com/repos/buildpacks/pack/releases/26958076/assets{?name,label}",
-    "html_url": "https://github.com/buildpacks/pack/releases/tag/v0.11.0",
-    "id": 26958076,
-    "node_id": "MDc6UmVsZWFzZTI2OTU4MDc2",
-    "tag_name": "v0.11.0",
-    "target_commitish": "release/0.11.0",
-    "name": "pack v0.11.0",
+    "url": "https://api.github.com/repos/buildpacks/pack/releases/27469720",
+    "assets_url": "https://api.github.com/repos/buildpacks/pack/releases/27469720/assets",
+    "upload_url": "https://uploads.github.com/repos/buildpacks/pack/releases/27469720/assets{?name,label}",
+    "html_url": "https://github.com/buildpacks/pack/releases/tag/v0.11.2",
+    "id": 27469720,
+    "node_id": "MDc6UmVsZWFzZTI3NDY5NzIw",
+    "tag_name": "v0.11.2",
+    "target_commitish": "release/0.11.2",
+    "name": "pack v0.11.2",
     "draft": false,
     "author": {
       "login": "github-actions[bot]",
@@ -32,14 +32,14 @@
       "site_admin": false
     },
     "prerelease": false,
-    "created_at": "2020-05-27T17:51:17Z",
-    "published_at": "2020-05-27T20:32:39Z",
+    "created_at": "2020-06-11T19:33:27Z",
+    "published_at": "2020-06-11T20:39:24Z",
     "assets": [
       {
-        "url": "https://api.github.com/repos/buildpacks/pack/releases/assets/21112685",
-        "id": 21112685,
-        "node_id": "MDEyOlJlbGVhc2VBc3NldDIxMTEyNjg1",
-        "name": "pack-v0.11.0-linux.tgz",
+        "url": "https://api.github.com/repos/buildpacks/pack/releases/assets/21669053",
+        "id": 21669053,
+        "node_id": "MDEyOlJlbGVhc2VBc3NldDIxNjY5MDUz",
+        "name": "pack-v0.11.2-linux.tgz",
         "label": "",
         "uploader": {
           "login": "github-actions[bot]",
@@ -63,17 +63,17 @@
         },
         "content_type": "application/gzip",
         "state": "uploaded",
-        "size": 6043552,
-        "download_count": 363,
-        "created_at": "2020-05-27T17:58:35Z",
-        "updated_at": "2020-05-27T17:58:35Z",
-        "browser_download_url": "https://github.com/buildpacks/pack/releases/download/v0.11.0/pack-v0.11.0-linux.tgz"
+        "size": 6043807,
+        "download_count": 20,
+        "created_at": "2020-06-11T19:40:29Z",
+        "updated_at": "2020-06-11T19:40:29Z",
+        "browser_download_url": "https://github.com/buildpacks/pack/releases/download/v0.11.2/pack-v0.11.2-linux.tgz"
       },
       {
-        "url": "https://api.github.com/repos/buildpacks/pack/releases/assets/21112684",
-        "id": 21112684,
-        "node_id": "MDEyOlJlbGVhc2VBc3NldDIxMTEyNjg0",
-        "name": "pack-v0.11.0-macos.tgz",
+        "url": "https://api.github.com/repos/buildpacks/pack/releases/assets/21669052",
+        "id": 21669052,
+        "node_id": "MDEyOlJlbGVhc2VBc3NldDIxNjY5MDUy",
+        "name": "pack-v0.11.2-macos.tgz",
         "label": "",
         "uploader": {
           "login": "github-actions[bot]",
@@ -97,17 +97,17 @@
         },
         "content_type": "application/gzip",
         "state": "uploaded",
-        "size": 6294528,
-        "download_count": 476,
-        "created_at": "2020-05-27T17:58:34Z",
-        "updated_at": "2020-05-27T17:58:34Z",
-        "browser_download_url": "https://github.com/buildpacks/pack/releases/download/v0.11.0/pack-v0.11.0-macos.tgz"
+        "size": 6295100,
+        "download_count": 15,
+        "created_at": "2020-06-11T19:40:28Z",
+        "updated_at": "2020-06-11T19:40:29Z",
+        "browser_download_url": "https://github.com/buildpacks/pack/releases/download/v0.11.2/pack-v0.11.2-macos.tgz"
       },
       {
-        "url": "https://api.github.com/repos/buildpacks/pack/releases/assets/21112686",
-        "id": 21112686,
-        "node_id": "MDEyOlJlbGVhc2VBc3NldDIxMTEyNjg2",
-        "name": "pack-v0.11.0-windows.zip",
+        "url": "https://api.github.com/repos/buildpacks/pack/releases/assets/21669055",
+        "id": 21669055,
+        "node_id": "MDEyOlJlbGVhc2VBc3NldDIxNjY5MDU1",
+        "name": "pack-v0.11.2-windows.zip",
         "label": "",
         "uploader": {
           "login": "github-actions[bot]",
@@ -131,16 +131,16 @@
         },
         "content_type": "application/zip",
         "state": "uploaded",
-        "size": 6069063,
-        "download_count": 90,
-        "created_at": "2020-05-27T17:58:35Z",
-        "updated_at": "2020-05-27T17:58:36Z",
-        "browser_download_url": "https://github.com/buildpacks/pack/releases/download/v0.11.0/pack-v0.11.0-windows.zip"
+        "size": 6068982,
+        "download_count": 13,
+        "created_at": "2020-06-11T19:40:30Z",
+        "updated_at": "2020-06-11T19:40:30Z",
+        "browser_download_url": "https://github.com/buildpacks/pack/releases/download/v0.11.2/pack-v0.11.2-windows.zip"
       }
     ],
-    "tarball_url": "https://api.github.com/repos/buildpacks/pack/tarball/v0.11.0",
-    "zipball_url": "https://api.github.com/repos/buildpacks/pack/zipball/v0.11.0",
-    "body": "# pack v0.11.0\r\n> This is a **beta** release of the Cloud Native Buildpack local CLI. This platform implementation should be relatively stable and reliable, but breaking changes in the underlying [specification](https://github.com/buildpack/spec) may be implemented without notice. Note that pack is intended for local image builds, and thus requires a Docker daemon. The [lifecycle](https://github.com/buildpack/lifecycle) should be used directly when building on cloud platforms.\r\n\r\n## Prerequisites\r\n\r\n- The [Docker daemon](https://www.docker.com/get-started) must be installed on your workstation or accessible over the network.\r\n\r\n## Install\r\n\r\n#### macOS\r\n\r\nIf you're on macOS, you can use Homebrew:\r\n\r\n```bash\r\n$ brew install buildpacks/tap/pack\r\n```\r\n\r\n#### Linux\r\n\r\nOn Linux you can use the one of the following methods of installation.\r\n\r\n##### Arch Linux\r\n\r\n- [pack-cli](https://aur.archlinux.org/packages/pack-cli/)\r\n- [pack-cli-bin](https://aur.archlinux.org/packages/pack-cli-bin/)\r\n\r\n##### Command\r\n\r\n```bash\r\n(export GH=buildpacks/pack; export LATEST=$(curl -s https://api.github.com/repos/$GH/releases/latest | grep -o -E \"https://.+?-linux.tgz\" | head -1); echo \"$LATEST\"; curl -sSL \"$LATEST\" | sudo tar -C /usr/local/bin/ --no-same-owner -xzv $(basename $GH))\r\n```\r\n\r\n#### Others\r\n\r\nOtherwise:\r\n\r\n1. Download the `.tgz` or `.zip` file for your platform\r\n2. Extract the `pack` binary\r\n3. (Optional) Add the directory containing `pack` to `PATH`, or copy `pack` to a directory like `/usr/local/bin`\r\n\r\n## Run\r\n\r\nRun the command `pack`.\r\n\r\nYou should see the following output\r\n\r\n```text\r\nUsage:\r\n  pack [command]\r\n\r\nAvailable Commands:\r\n  build                 Generate app image from source code\r\n  rebase                Rebase app image with latest run image\r\n  inspect-image         Show information about a built image\r\n  set-run-image-mirrors Set mirrors to other repositories for a given run image\r\n  set-default-builder   Set default builder used by other commands\r\n  inspect-builder       Show information about a builder\r\n  suggest-builders      Display list of recommended builders\r\n  trust-builder         Trust builder\r\n  create-builder        Create builder image\r\n  package-buildpack     Package buildpack in OCI format.\r\n  suggest-stacks        Display list of recommended stacks\r\n  version               Show current 'pack' version\r\n  report                Display useful information for reporting an issue\r\n  completion            Outputs completion script location\r\n  help                  Help about any command\r\n\r\nFlags:\r\n  -h, --help         Help for 'pack'\r\n      --no-color     Disable color output\r\n  -q, --quiet        Show less output\r\n      --timestamps   Enable timestamps in output\r\n  -v, --verbose      Show more output\r\n      --version      Show current 'pack' version\r\n\r\nUse \"pack [command] --help\" for more information about a command.\r\n```\r\n\r\n## Info\r\n\r\nBuilders created with this release of the pack CLI continue to contain [lifecycle v0.7.5](https://github.com/buildpack/lifecycle/releases/tag/v0.7.5) by default.\r\n\r\n## Features\r\n\r\n* Performance and security improvements ([learn more](https://medium.com/buildpacks/faster-more-secure-builds-with-pack-0-11-0-4d0c633ca619))\r\n    * Trusted builders (#634, #644)\r\n    * Run `analyze`, `restore`, and `export` via a trusted lifecycle image (#631)\r\n    * Run `creator` from the lifecycle (#595)\r\n* Hide stack mixins for `inspect-builder` behind `--verbose` option (#630)\r\n* Add `.cnb` file support for `package-buildpack` dependencies (#633)\r\n* Support `.cnb` files in `create-builder` (#611)\r\n* Overwrite permissions in zip files for FAT formatted entries (#598)\r\n* Error when `--publish` and `--no-pull` are used in an unsupported fashion (#594)\r\n* Adds experimental flagging (#576)\r\n* Add Google builder to `suggest-builders` (#637)\r\n* Update `suggest-builders` to use Paketo builders instead of Cloud Foundry builders (#559)\r\n* Support windows containers for `create-builder` [experimental] (#571)\r\n* Support buildpacks from buildpack registry for `create-builder` and `build` [experimental] (#540)\r\n\r\n## Fixes\r\n\r\n* Propagate verbose logging flag to `analyze` and `build` phases (#636)\r\n* `build ... --publish` now respects default process type (#602)\r\n* Support substitution for config files such as from `envsubst` (#592)\r\n* Remove daemon access from `restore` phase (#589)\r\n* `create-builder` can now handle when both `uri` and `image` buildpacks as provided (#625)\r\n\r\n## Breaking Changes\r\n\r\n*  Network mode is now respected for all phases of the lifecycle (#595)\r\n    * This means that publishing (`--publish`) might be impacted if the network mode does not have access to the registry to which the image is being published to. Previously, `analyze` and `export` phases used `host` network mode when publishing and now use the `default` (aka `bridge`) network mode.\r\n    * Related conversations: [Slack](https://buildpacks.slack.com/archives/CJ6B92ZSB/p1586528175087900)\r\n    * Solution:\r\n        * Network mode can be configured via the `--network` flag.\r\n* Builders with lifecycle versions that don't have a published lifecycle image will error (#644)\r\n    * This would only impacts builders that are not trusted.\r\n    * Solutions:\r\n        * Set builder as trusted via `pack trust-builder ...` or pass `--trust-builder` to the `build` command.\r\n        * Update builder to use the latest versions of supported lifecycle versions."
+    "tarball_url": "https://api.github.com/repos/buildpacks/pack/tarball/v0.11.2",
+    "zipball_url": "https://api.github.com/repos/buildpacks/pack/zipball/v0.11.2",
+    "body": "# pack v0.11.2\r\n> This is a **beta** release of the Cloud Native Buildpack local CLI. This platform implementation should be relatively stable and reliable, but breaking changes in the underlying [specification](https://github.com/buildpack/spec) may be implemented without notice. Note that pack is intended for local image builds, and thus requires a Docker daemon. The [lifecycle](https://github.com/buildpack/lifecycle) should be used directly when building on cloud platforms.\r\n\r\n## Prerequisites\r\n\r\n- The [Docker daemon](https://www.docker.com/get-started) must be installed on your workstation or accessible over the network.\r\n\r\n## Install\r\n\r\n#### macOS\r\n\r\nIf you're on macOS, you can use Homebrew:\r\n\r\n```bash\r\n$ brew install buildpacks/tap/pack\r\n```\r\n\r\n#### Linux\r\n\r\nOn Linux you can use the one of the following methods of installation.\r\n\r\n##### Arch Linux\r\n\r\n- [pack-cli](https://aur.archlinux.org/packages/pack-cli/)\r\n- [pack-cli-bin](https://aur.archlinux.org/packages/pack-cli-bin/)\r\n\r\n##### Command\r\n\r\n```bash\r\n(export GH=buildpacks/pack; export LATEST=$(curl -s https://api.github.com/repos/$GH/releases/latest | grep -o -E \"https://.+?-linux.tgz\"); echo \"$LATEST\"; curl -sSL \"$LATEST\" | sudo tar -C /usr/local/bin/ --no-same-owner -xzv $(basename $GH))\r\n```\r\n\r\n#### Others\r\n\r\nOtherwise:\r\n\r\n1. Download the `.tgz` or `.zip` file for your platform\r\n2. Extract the `pack` binary\r\n3. (Optional) Add the directory containing `pack` to `PATH`, or copy `pack` to a directory like `/usr/local/bin`\r\n\r\n## Run\r\n\r\nRun the command `pack`.\r\n\r\nYou should see the following output\r\n\r\n```text\r\nUsage:\r\n  pack [command]\r\n\r\nAvailable Commands:\r\n  build                 Generate app image from source code\r\n  rebase                Rebase app image with latest run image\r\n  inspect-image         Show information about a built image\r\n  set-run-image-mirrors Set mirrors to other repositories for a given run image\r\n  set-default-builder   Set default builder used by other commands\r\n  inspect-builder       Show information about a builder\r\n  suggest-builders      Display list of recommended builders\r\n  trust-builder         Trust builder\r\n  create-builder        Create builder image\r\n  package-buildpack     Package buildpack in OCI format.\r\n  suggest-stacks        Display list of recommended stacks\r\n  version               Show current 'pack' version\r\n  report                Display useful information for reporting an issue\r\n  completion            Outputs completion script location\r\n  help                  Help about any command\r\n\r\nFlags:\r\n  -h, --help         Help for 'pack'\r\n      --no-color     Disable color output\r\n  -q, --quiet        Show less output\r\n      --timestamps   Enable timestamps in output\r\n  -v, --verbose      Show more output\r\n      --version      Show current 'pack' version\r\n\r\nUse \"pack [command] --help\" for more information about a command.\r\n```\r\n\r\n## Info\r\n\r\nBuilders created with this release of the pack CLI continue to contain [lifecycle v0.7.5](https://github.com/buildpack/lifecycle/releases/tag/v0.7.5) by default.\r\n\r\n## Features\r\n\r\n\r\n## Fixes\r\n\r\n* Use lifecycle image instead of builder when exporting to daemon and builder is untrusted (#684)"
   },
   "repository": {
     "id": 138544269,

